packages/block-library/src/embed/transforms.js

```diff
@@ -33,10 +33,25 @@ const transforms = {
 		{
 			type: 'block',
 			blocks: [ 'core/paragraph' ],
+			isMatch: ( { url } ) => !! url,
 			transform: ( { url, caption } ) => {
-				const link = <a href={ url }>{ caption || url }</a>;
+				let linkText = url;
+				if ( caption?.trim() ) {
+					const captionEl = create( { html: caption } );
+					linkText = toHTMLString( {
```

Comment:
@ellatrix am I doing some kind of circles here with `create and toHTMLString`? 🤔

Reply:
Why do we want add the URL at all? Why merge two things and potentially cause data loss? Maybe create two paragraphs, a link and the caption?

Reply:
You're right about the data loss. I've updated to create two paragraphs.

Reply:
I'm not sure what would be a better setup, but it I think we should definitely avoid linking the caption.

Alternatives:

* A paragraph with the URL, a `<br>` element, and the caption. The nice thing is that it remains a single block.
* An version of the embed block that just displaces the URL text instead of the embed.